int32 id
float64 range
float64 phi